src/py_string_cache.rs

```diff
@@ -183,3 +184,23 @@ impl PyStringCache {
         self.entries.fill(None);
     }
 }
+
+pub fn pystring_fast_new<'py>(py: Python<'py>, s: &str, ascii_only: bool) -> Bound<'py, PyString> {
+    if ascii_only {
+        unsafe { pystring_ascii_new(py, s) }
+    } else {
+        PyString::new_bound(py, s)
+    }
+}
+
+/// Faster creation of PyString from an ASCII string, inspired by
+/// https://github.com/ijl/orjson/blob/3.10.0/src/str/create.rs#L41
+unsafe fn pystring_ascii_new<'py>(py: Python<'py>, s: &str) -> Bound<'py, PyString> {
+    let ptr = ffi::PyUnicode_New(s.len() as isize, 127);
+    // see https://github.com/pydantic/jiter/pull/72#discussion_r1545485907
+    debug_assert_eq!(ffi::PyUnicode_KIND(ptr), ffi::PyUnicode_1BYTE_KIND);
+    let data_ptr = ffi::PyUnicode_DATA(ptr) as *mut u8;
```

Comment:
I slightly prefer `.cast()` to `as` conversions, mostly because `.cast()` doesn't convert const to mut accidentally so it's a good habit to get into.

```suggestion
    let data_ptr = ffi::PyUnicode_DATA(ptr).cast();
```